sqlite: update to 3.38.0, fossil: update to 2.18.

ologantr opened the pull request:
Also remove -DSQLITE_ENABLE_JSON1 as JSON is now on by default.
https://sqlite.org/releaselog/3_38_0.html

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture (x86_64-glibc)

mobinmob commented:
`fossil` update depends on the sqlite update.

ologantr commented:
> `fossil` update to 2.18 depends on the sqlite update.

I was actually waiting on this being merged.
Should I push a commit for fossil on this 
branch and edit this pull request?

mobinmob commented:
If that is what you want :)
As a sidenote: In my tests fossil seems to need both `--disable-internal-sqlite` and -`-with-sqlite="${XBPS_CROSS_BASE}/usr/include/"` in order to use the system sqlite. I may be mistaken, but check it out.

ologantr commented:
> If that is what you want :) As a sidenote: In my tests fossil seems to need both `--disable-internal-sqlite` and -`-with-sqlite="${XBPS_CROSS_BASE}/usr/include/"` in order to use the system sqlite. I may be mistaken, but check it out.

Yes, you're right: without --with-sqlite, configure
falls back to the fossil-provided one even if
--disable-internal-sqlite was used. I've
updated fossil and added this switch.
